pkg/kfapp/gcp/replicate_image.go

```diff
@@ -0,0 +1,204 @@
+package gcp
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/kubeflow/kfctl/v3/pkg/kfapp/kustomize"
+	log "github.com/sirupsen/logrus"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const INPUT_IMAGE = "inputImage"
+const OUTPUT_IMAGE = "outputImage"
+const TASK_NAME = "images-replication"
+const KUSTOMIZE_FOLDER = "kustomize"
+
+// buildContext: gs://<GCS bucket>/<path to .tar.gz>
+func GenerateReplicationPipeline(registry string, buildContext string, include string, exclude string,
+	outputFileName string) error {
+	replicateTasks := make(map[string]string)
+
+	infos, err := ioutil.ReadDir(".")
+	if err != nil {
+		return err
+	}
+	foundKus := false
+	for _, info := range infos {
+		if info.IsDir() && info.Name() == KUSTOMIZE_FOLDER {
+			foundKus = true
+			break
+		}
+	}
+	if !foundKus {
+		return fmt.Errorf("kustomize folder not found, have you executed kfctl build yet?")
+	}
+	err = filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			absPath, err := filepath.Abs(path)
+			if err != nil {
+				return err
+			}
+
+			kustomizationFilePath := filepath.Join(absPath, "kustomization.yaml")
+			if _, err := os.Stat(kustomizationFilePath); err == nil {
+				kustomization := kustomize.GetKustomization(absPath)
+				for i, image := range kustomization.Images {
+					curName := image.Name
+					if image.NewName != "" {
+						curName = image.NewName
+					}
+					// check exclude first
+					if strings.HasPrefix(curName, exclude) {
+						log.Infof("image %v matches exclude prefix %v, skipping\n", curName, exclude)
+						continue
+					}
+					// then check include
+					if include != "" && (!strings.HasPrefix(curName, include)) {
+						log.Infof("image %v doesn't match include prefix %v, skipping\n", curName, include)
+						continue
+					}
+					newName := strings.Join([]string{registry, image.Name}, "/")
+
+					if (image.NewTag == "") == (image.Digest == "") {
+						log.Warnf("One and only one of NewTag or Digest can exist for image %s, skipping\n",
+							image.Name)
+						continue
+					}
+
+					if image.NewTag != "" {
+						replicateTasks[strings.Join([]string{image.Name, image.NewTag}, ":")] =
+							strings.Join([]string{newName, image.NewTag}, ":")
+					}
+					if image.Digest != "" {
+						replicateTasks[strings.Join([]string{image.Name, image.Digest}, "@")] =
+							strings.Join([]string{newName, image.Digest}, "@")
+					}
+
+					log.Infof("Replacing image name from %s to %s", image.Name, newName)
+					kustomization.Images[i].NewName = newName
+				}
+			}
+			return nil
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	taskIns := pipeline.Task{
```

Comment:
Do you need to generate the Task using go? Can't we just provide people with a YAML file in the repo?